src/app/modules/courses/components/course-card/course-card.component.spec.ts

```diff
@@ -14,10 +16,23 @@ describe('CourseCardComponent', () => {
 
     fixture = TestBed.createComponent(CourseCardComponent);
     component = fixture.componentInstance;
+    component.course = {...coursesMock[0]};
+    el = fixture.nativeElement;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should handle actions', fakeAsync (() => {
+    spyOn(component.edit, 'emit');
+    spyOn(component.delete, 'emit');
+    const actionPanel = el.querySelector('.action');
+    const buttons = actionPanel.querySelectorAll('button');
+    buttons.forEach(btn => btn.click());
```

Comment:
such tests quite hard to read. It is better to split it on 2 different tests. Also I am not sure, that tick is needed here

Reply:
tick needed to recognize  angular that click event emitted

Reply:
divided